giantswarm/happa
# [CP Auth 5/5]: Add React components and link everything together

Closes https://github.com/giantswarm/giantswarm/issues/8622

This PR includes the React components that make the CP Auth usable in the app.

Note: You might notice that after you login/logout of the CP API, happa will do a hard refresh. That is because app catalogs are loaded as part of happa's initial render, and doing a hard refresh is an easy way to trigger that step again.

### Preview

<details>
<summary>User dropdown</summary>

![image](https://user-images.githubusercontent.com/13508038/90109833-a55fe680-dd4c-11ea-8c18-84b4a4e174ef.png)

</details>

<details>
<summary>Control plane access page - logged out</summary>

![image](https://user-images.githubusercontent.com/13508038/90109893-b90b4d00-dd4c-11ea-9eee-42f653d4d7ec.png)

</details>

<details>
<summary>Control plane access page - logged in</summary>

![image](https://user-images.githubusercontent.com/13508038/90110029-ea841880-dd4c-11ea-8318-c9c14e9b72eb.png)

</details>

### How to test

1. Deploy this branch to a testing installation.

<details>
<summary>2. Click on the user drop-down in the top right of the page. Press on the `Control Plane Access` button.</summary>

![image](https://user-images.githubusercontent.com/13508038/84696202-667c0300-af4c-11ea-95b5-a29226b7ab19.png)

</details>

<details>
<summary>3. Click on the 'Log in to Control Plane via OIDC' button</summary>

![image](https://user-images.githubusercontent.com/13508038/90109535-41d5b900-dd4c-11ea-9f4b-f154b532d940.png)

</details>

4. You will be redirected to GitHub. Authorize the app.

5. You will be redirected back to the Happa homepage.

<details>
<summary>6.. Enter the 'App Catalogs' page (top menu). The app catalogs are fetched using the Control Plane API. If you are doubting that, you can inspect the network tab in DevTools to see that there is still hope in this world.</summary>

![image](https://user-images.githubusercontent.com/13508038/84696424-c5da1300-af4c-11ea-9f84-57e0f08b7494.png)

</details>
